Python/compile.c

```diff
@@ -7500,16 +7500,42 @@ assemble_emit_exception_table_item(struct assembler *a, int value, int msb)
     write_except_byte(a, (value&0x3f) | msb);
 }
 
+/* Make room for at least (logical_length+to_add)*unitsize in the
+   bytes object. Use exponential growth for O(1) amortized runtime. */
+static int
+bytes_make_room(PyObject **bytes, int logical_length,
+                int to_add, Py_ssize_t unitsize)
+{
+    // Make sure we can successfully do the addition.
+    if (logical_length > INT_MAX - to_add) {
+        PyErr_NoMemory();
+        return 0;
+    }
+    // The existing logical buffer should always fit in a Py_ssize_t
+    assert(logical_length <= PY_SSIZE_T_MAX / unitsize);
+    Py_ssize_t b_len = PyBytes_GET_SIZE(*bytes);
+    if (unitsize * logical_length >= b_len - to_add * unitsize) {
+        // There's not enough room. Double it.
+        if (b_len > PY_SSIZE_T_MAX / 2) {
```

Comment:
This has the same issue as the backport.
We want to be able to index all code object structures with an `int`, so you'll need `INT_MAX` instead of `PY_SSIZE_T_MAX` and it should be an overflow error, not a memory error.

Reply:
To clarify, we want `a_bytecode` to be able to have `INT_MAX` code units, not just `INT_MAX` bytes, right?

Reply:
We probably want the smaller amount to avoid risk of overflow. So the length in *bytes* should be less than `INT_MAX`.